internal/core/services/hash_service.go

```diff
@@ -0,0 +1,175 @@
+package services
```

Comment:
If we already named package `services`, we shouldn't use `_service` postfix in file names, so I'm advising to remove from file names postfixes `_service` and `_ports`

Reply:
Fixed, renamed.